tests/devices/i04/test_transfocator.py

```diff
@@ -1,66 +1,74 @@
 
 
-def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
-    fake_transfocator: Transfocator,
+async def set_beamsize_to_same_value_as_mock_signal(
+    transfocator: Transfocator, value: float
 ):
-    fake_transfocator.beamsize_set_microns.sim_put(100)  # type: ignore
-    set_status = fake_transfocator.set(100)
-    set_status.wait(0.01)
-    assert set_status.done and set_status.success
+    set_mock_value(transfocator.beamsize_set_microns, value)
+    await transfocator.set(value)
 
 
-@patch("dodal.devices.i04.transfocator.sleep")
-def test_when_beamsize_set_then_set_correctly_on_device_and_waited_on(
-    mock_sleep,
+@patch("dodal.devices.i04.transfocator.Transfocator._observe_beamsize_microns")
+async def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
+    mock_observe_beamsize_microns,
     fake_transfocator: Transfocator,
 ):
-    given_predicted_lenses_is_half_of_beamsize(fake_transfocator)
-
-    transfocator_statuses = SubscriptionStatus(
-        fake_transfocator.predicted_vertical_num_lenses,
-        lambda *, old_value, value, **kwargs: value == 157,
-    )
-    transfocator_statuses &= SubscriptionStatus(
-        fake_transfocator.number_filters_sp,
-        lambda *, old_value, value, **kwargs: value == 157,
-    )
-    transfocator_statuses &= SubscriptionStatus(
-        fake_transfocator.start,
-        lambda *, old_value, value, **kwargs: value == 1,
+    await asyncio.wait_for(
+        set_beamsize_to_same_value_as_mock_signal(fake_transfocator, 100.0),
+        timeout=0.01,
     )
+    mock_observe_beamsize_microns.assert_not_awaited()
+
 
+@patch("dodal.devices.i04.transfocator.wait_for_value")
+async def test_when_beamsize_set_then_set_correctly_on_device_and_waited_on(
+    mock_wait_for_value,
+    fake_transfocator: Transfocator,
+):
+    given_predicted_lenses_is_half_of_beamsize(fake_transfocator)
     set_status = fake_transfocator.set(315)
 
-    transfocator_statuses.wait(0.1)
     assert not set_status.done
 
-    fake_transfocator.start_rbv.get = MagicMock(side_effect=[1, 0])
+    await asyncio.gather(
+        wait_for_value(fake_transfocator.predicted_vertical_num_lenses, 157, 0.1),
+        wait_for_value(fake_transfocator.number_filters_sp, 157, 0.1),
+        wait_for_value(fake_transfocator.start, 1, 0.1),
+    )
 
-    set_status.wait(0.1)
+    await set_status
     assert set_status.done and set_status.success
 
-    try:
-        transfocator_statuses.wait(0.1)
-    except BaseException:
-        pass
+
+async def test_if_timeout_exceeded_and_start_rbv_not_equal_to_set_value_then_timeout_exception(
+    fake_transfocator: Transfocator,
+) -> None:
+    with patch.object(fake_transfocator, "TIMEOUT", 0):
+        given_predicted_lenses_is_half_of_beamsize(fake_transfocator)
+        fake_transfocator.start.get_value = AsyncMock(side_effect=[0, 1])
+        with pytest.raises(TimeoutError):
```

Comment:
This test fails in python 3.10 as, in this version, the standard TimeoutError is different to asyncio.TimeoutError. Doing `with pytest.raises((TimeoutError, asyncio.exceptions.TimeoutError)):` should make it work for all versions